### ISSUE TITLE
Potential chart affected by zooming during neuron stimulation 

### ISSUE THREAD

**Issue by elisemorgan** [phetsims/neuron#109]
Test device: iPad 3, Disessa

Operating System: iOS 9.2 13C75

Browser: Safari

Problem description: The potential plot is sharper if the zoom is changed while the neuron is stimulated. In the screenshot below, the first peak is with zoom and the second peak is normal. 

Steps to reproduce: Select potential chart. Stimulate neuron and zoom in and out during stimulation. 

Severity:

Screenshots: 
![image](https://cloud.githubusercontent.com/assets/12610678/12021337/0b87976c-ad43-11e5-8fe5-896f1515b195.png)

Troubleshooting information (do not edit):
Name: ‪Neuron‬
URL: http://www.colorado.edu/physics/phet/dev/html/neuron/1.0.0-rc.1/neuron_en.html
Version: 1.0.0-rc.1 2015-12-22 23:06:19 UTC
Features missing: fullscreen
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (iPad; CPU OS 9_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13C75 Safari/601.1
Language: en-us
Window: 1024x672
Pixel Ratio: 2/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 IMGSGX543-124.1)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.00)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 8 uniform: 128
Texture: size: 4096 imageUnits: 8 (vertex: 8, combined: 8)
Max viewport: 4096x4096
OES_texture_float: true
Dependencies JSON: {"assert":{"sha":"d4909677","branch":"HEAD"},"axon":{"sha":"6f3c2e01","branch":"HEAD"},"babel":{"sha":"9618fad2","branch":"master"},"brand":{"sha":"c9282935","branch":"HEAD"},"chipper":{"sha":"1d1d5d32","branch":"HEAD"},"dot":{"sha":"7a28b1a9","branch":"HEAD"},"griddle":{"sha":"3adb78bd","branch":"HEAD"},"joist":{"sha":"191b15a4","branch":"HEAD"},"kite":{"sha":"4a947119","branch":"HEAD"},"neuron":{"sha":"3b7eee9d","branch":"HEAD"},"phet-core":{"sha":"73a034c6","branch":"HEAD"},"phetcommon":{"sha":"d712ace0","branch":"HEAD"},"scenery":{"sha":"47143cdd","branch":"HEAD"},"scenery-phet":{"sha":"a7f975d1","branch":"HEAD"},"sherpa":{"sha":"be8c4fc9","branch":"HEAD"},"sun":{"sha":"64555735","branch":"HEAD"},"tandem":{"sha":"07fbe58c","branch":"HEAD"}}

**Comment by elisemorgan** [comment]
*Not* reproducible on iPad 2 w/ iOS 9

**Comment by arouinfar** [comment]
Also not reproducible on the latest desktop version of Safari.

**Comment by jbphet** [comment]
Also not reproducible on Win7 + Chrome, Win7 + Firefix, or Win7 + IE11.

Assigning to @elisemorgan to see if she can reproduce on any other devices besides iPad 3.

**Comment by amanda-phet** [comment]
I could not reproduce on OSX 10.10 Safari.

**Comment by amyh-phet** [comment]
Not reproducible on Chromebook either.

**Comment by elisemorgan** [comment]
@jbphet This issue appears to be iPad 3 specific

**Comment by arouinfar** [comment]
I noticed this issue on Tycho too -- iPad2 running iOS 9.2.

**Comment by arouinfar** [comment]
I just saw @elisemorgan's comment that she couldn't reproduce on iPad2/iOS9. The deformity in the graph is not as noticeable as the screenshots in the original post.

The first bump in the graph in the first screenshot below has the correct shape. There's a little kink around 2ms. The following three times I stimulated the neuron, I wiggled the zoom back and forth while the bump was being drawn. In all three cases, the graph is a bit deformed. The lack of the initial kink is the most noticeable difference. 

![neuron1](https://cloud.githubusercontent.com/assets/8419308/12043203/f6674c62-ae3f-11e5-8682-065b3af41d90.png)
![neuron2](https://cloud.githubusercontent.com/assets/8419308/12043204/f83f816c-ae3f-11e5-88a8-446ceb2dceab.png)

**Comment by jbphet** [comment]
I spent some time looking at this, and it is due to the sim trying to keep up while simultaneously working to re-scale everything in the view window.  There is no easy fix.  Some of the recent changes seem to help such that the very spiky graph that @elisemorgan was able to create is difficult if not impossible to reproduce, but some loss of fidelity is still occurring when this test is run on the latest RC (1.0.0-rc.2).  This is a rare and kind of abusive use case, and in my opinion this issue should not block publication.  Assigning to @ariel-phet to disposition.

**Comment by ariel-phet** [comment]
Does not occur with regular zooming in and out. Closing

**Comment by jbphet** [comment]
A related issue (see #114) has pushed this over the threshold to the point where I will likely add some code to address it, at least partially.  Reopening.

**Comment by jbphet** [comment]
After more investigation, I realized that having the sim in fast forward mode is vital to reproducing this issue.  While doing that, I was able to reproduce the problem on an iPad 2 running iOS 9.2.1 (the PhET test device named "Witten").  Screenshot:

![neuron-ap-while-zooming-old](https://cloud.githubusercontent.com/assets/3082831/12930426/85436180-cf35-11e5-9869-d038f578b1af.png)

I added code to break large simulation time ticks into multiple smaller ticks and re-ran the test.  The resulting graph is much improved.  Screenshot:

![neuron-ap-while-zooming-fixed](https://cloud.githubusercontent.com/assets/3082831/12930464/b0845070-cf35-11e5-8c40-bf621f759ca4.png)
